plugin-testlib/src/main/kotlin/io/spine/tools/gradle/testing/Replacement.kt

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2021, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.gradle.testing
+
+import java.io.File
+
+/**
+ * A replacement to be made in a text.
+ *
+ * Scans the text for the occurrences of the formatted token, such as `@MY_TOKEN@`,
+ * and replaces them with the passed text value.
+ *
+ * @param token the name of the token, without `@` symbols
+ * @param value the value to replace the token occurrences to; may be blank
+ */
+public class Replacement(token: String, public val value: String) {
+
+    private val token: String
+
+    init {
+        require(token.isNotBlank())
+        this.token = token
+    }
+
+    /**
+     * Returns the token formatted as used in the text, in which the replacement will take place.
+     *
+     * Example:
+     * ```
+     *      Replacement("VERSION", ...).token() // -> returns "@VERSION@"
+     * ```
+     */
+    internal fun token(): String {
+        return "@$token@"
+    }
+
+    /**
+     * Replaces all occurrences of [token][token].
+     *
+     * If the passed [file] is a folder, the occurrences are replaced in all files which reside
+     * in this folder and its sub-folders recursively.
+     *
+     * Optionally, allows to specify the folder, which files should be excluded from replacement.
+     */
+    public fun replaceIn(file: File, excludeFolder: File? = null) {
```

Comment:
I think that the logic of exclusion should be responsibility of the `GradleProject` class. The `Replacement` class currently feels as a bit too “greedy” for responsibilities. 

The logic of having (or not having) `buildSrc` is the responsibility of `GradleProject`. I guess, it should walk the `Replacement` all over around giving it `File`s to deal with. 

Now we exclude only `buildSrc`, but there may be other cases of filtering out other directories or files (e.g. using `Predicate`s over them). If we need this, we'd have to modify only `GradleProject`. The current design would force us to modify both `GradleProject` and `Replacement`.